content/blog/2022-03-22-type-alias/index.md

```diff
@@ -0,0 +1,210 @@
++++
+title = "Type-based Alias Analysis"
+[extra]
+latex = true
+[[extra.authors]]
+name = "Andrew Butt"
+link = "TODO"
+[[extra.authors]]
+name = "Andrey Yao"
+link = "https://www.youtube.com/watch?v=dQw4w9WgXcQ"
++++
+
+## Introduction
+The type system of a statically-typed language allows compilers to reject illegal programs during the type-checking stage of compilation. In a sense, the typing information attached to variables, functions, etc. is a refinement on the set of valid program states, which can be approximated without ever running the program. Although programming language types are often used to locate certain errors during compile time, it's not their only use.
+
+"Type-Based Alias Analysis"<sup>[1]</sup> by Diwan, McKinley, and Moss examines how the same principal can be applied to alias analysis, a type of conservative analysis that determines whether two given pointer variables might interfere with each other, i.e. pointing to the same memory address. Although there had been prior research on alias analysis, Diwan et al.'s type-based alias analysis (TBAA) has the following advantages:
```

Comment:
"i.e." should have a comma after it: https://capra.cs.cornell.edu/styleguide/#egie

But actually, TBH, I'd just rephrase "might interfere with each other, i.e. pointing to the same memory address" as plain "might point to the same memory address." That's a little more precise.